### DISCUSSION TITLE
Voice Navigation Issue

### DISCUSSION THREAD

**Discussion by archer46** [osmandapp/OsmAnd#13958] [Q&A] [answered]
Hello, not sure what is going on with OsmAnd for me but I can't get voice navigation to work. Considering how outdated the posts of other people experiencing this issue are I'm going to guess it's isolated to me. Everything looks right in my settings, so I don't know what's happening. I followed along with the instructions linked below to no success. I'm running stock Motorola android.

https://docs.osmand.net/en/main@latest/osmand/troubleshooting/navigation#voice-navigation

**Comment by Lee-Carre** [comment]
@archer46

Care to describe the exact problem (with reproduction steps), and what you've tried so far?

Pointers to related discussions / issues (sounds like you found some) might help, too.

**Comment by archer46** [comment, marked as answer]
@Lee-Carre 

Upon installing the app on a Moto G7 with absolutely no modifications to the stock settings of the app from Google Play Store, no audio was played by the app in any form whatsoever while navigating. I then ensured my settings matched those outlined in the previously linked article, and upon confirmation of this, restarting the app, and resuming navigating, no change occurred. The volume was up. Please let me know if there are any other details of use. Thank you for your reply!

**Reply by Lee-Carre** [reply]
Oh, complete failure. I see.

I assume that other apps have no problems playing audio via the same channel as you've set OsmAnd to use?

Presuming that you're using TTS, I trust that you have all the necessaries installed?
Have you tried with different engines, different voices?

What about pre-recorded voice prompts?

When you configured voice prompts, and when testing via navigation, were they both using the same profile? Almost all configuration is (by design) profile-specific. Though, with a mechanism to apply the current set to other profiles.

Which version of OsmAnd are you using?

**Reply by archer46** [reply]
@Lee-Carre 
I just investigated TTS further as I know next to nothing about this. I disabled the google speech services app long ago having no use for it, but apparently that is necessary for the TTS functionality. Re-enabling it resolved the issue. In that case I have a follow up question, what would you recommend for a non google dependent TTS?

**Reply by Lee-Carre** [reply]
Ah, good (read: bad) old Google trickery. I can well relate.

I was beginning to wonder if it was perhaps a Motorola snag of some kind.

I'm glad that you have success, now.
Side-note: thankyou for being rational & more concerned with resolving the problem than about fuzzy feels. Had a few too many cases (elsewhere) of hostility from butthurt (seemingly because I didn't kiss their ass, and was instead focused on the task), while simply cutting the BS by asking straight questions in order to help. No good deed goes unpunished, I guess.

Would you consider marking it as the answer, for the benefit of others in future?

Since I have background in user-experience design, what occurs to me is the silent failure (particularly of the OS or TTS system), instead of informing you that either TTS itself, or something TTS depends on, was disabled (or otherwise non-functional).
At least with a cryptic error message, one knows that there **is** a problem. Silence is, thus, worse.

As for non-(Google|proprietary) TTS (both engines & voices); I can't recommend [F-Droid](//www.f-droid.org/) strongly enough. There are a couple (non-outdated) engines available, with a variety of voices. You may wish to configure it with additional repositories (such as IzzyOnDroid's collection) to have greater variety.

However, some caveats:
* one of the more recent engines (‘R’-something) has trouble with OsmAnd (there's a few tickets about it, if you search by name)
* the fidelity of the voices currently available for other engines is not great (especially when your attention is elsewhere, like on the road); 80s synth-pop robo-voice flashbacks — so, depends if libre is important enough to accept lower fidelity — however, I gather that they could be much better if voice data was contributed; many current ones seem to be originally intended as demos

**Comment by archer46** [comment]
@Lee-Carre 
It just so happens I tried RHVoice just before receiving your reply and got stumped there myself! Thanks for pointing me in the direction of those bug reports, that got things working correctly. [This bug report](https://github.com/osmandapp/OsmAnd/issues/13672) in particular did the trick. You've been a massive help, I very much appreciate it.

**Reply by Lee-Carre** [reply]
Excellent; pleased to hear that.

Thankyou for the compliments, and the pointer.
